Sprint-1/implement/sum.test.js

```diff
@@ -1,36 +1,27 @@
-/* Sum the numbers in an array
-
-In this kata, you will need to implement a function that sums the numerical elements of an array
-
-E.g. sum([10, 20, 30]), target output: 60
-E.g. sum(['hey', 10, 'hi', 60, 10]), target output: 80 (ignore any non-numerical elements)
-*/
-
 const sum = require("./sum.js");
 
-// Acceptance Criteria:
-
-// Given an empty array
-// When passed to the sum function
-// Then it should return 0
-test.todo("given an empty array, returns 0")
+describe("sum", () => {
+  test("given an empty array, returns 0", () => {
+    expect(sum([])).toEqual(0);
+  });
 
-// Given an array with just one number
-// When passed to the sum function
-// Then it should return that number
+  test("given an array with just one number, returns that number", () => {
+    expect(sum([42])).toEqual(42);
+  });
 
-// Given an array containing negative numbers
-// When passed to the sum function
-// Then it should still return the correct total sum
+  test("given an array containing negative numbers, returns the correct total sum", () => {
+    expect(sum([-10, 20, -30, 40])).toEqual(20);
+  });
 
-// Given an array with decimal/float numbers
-// When passed to the sum function
-// Then it should return the correct total sum
+  test("given an array with decimal/float numbers, returns the correct total sum", () => {
+    expect(sum([1.5, 2.5, 3.5])).toEqual(7.5);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Note: Decimal numbers with `.5` in the decimal part happened to be able to be represented exactly in floating point number format.